PF-session3/pf3_assignment3/QuickSort.java

```diff
@@ -0,0 +1,69 @@
+
+/**
+ * This class has method to sort the array in ascending order.This code implements quick sort algorithm.
+ *
+ * @version 1 25 Jan 2018
+ * @author Akshita Dixit
+ */
+package pf3_assignment3;
+
+public class QuickSort {
+	
+	    /**
+	     * This method returns the index from where the array is one side smaller and 
+	     * other side greater than the pivot.
+	     * @param arr
+	     * @param first
+	     * @param last
+	     * @return
+	     */
+	    int partition(int arr[], int first, int last)
+	    {
+	        int pivot = arr[last]; 
+	        int iterator = (first-1);                                 // index of smaller element
+	        for (int counter=first; counter < last; counter++)
+	        {
+	                                                                 /* If current element is smaller than or
+	                                                                    equal to pivot*/
+	            if (arr[counter] <= pivot)
+	            {
+	                iterator++;
+	                int temp = arr[iterator];                        // swap arr[i] and arr[j]
+	                arr[iterator] = arr[counter];
```

Comment:
Common code should be extracted to methods. Here, we are repeating the code to swap numbers at two indices within the array.